internal/appconfig/remote.go

```diff
@@ -57,6 +57,47 @@ func FromRemoteApp(ctx context.Context, appName string) (*Config, error) {
 	}
 }
 
+func FromAppCompact(ctx context.Context, appCompact *api.AppCompact) (*Config, error) {
```

Comment:
Since both functions were using the same switch statement to get the config, I just moved that code to its own function that both `FromAppCompact` and `FromRemoteApp` can use. I opted not to have `FromRemoteApp` call `FromAppCompact` since `FromAppCompact` would have created another API client, which I figured could be avoided.